### PR TITLE
Make the settings in the lsp server nicer.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/lsp/server/server.toit [hunk 1]
```diff
@@ -53,9 +53,7 @@ CRASH-REPORT-RATE-LIMIT-MS ::= 30_000
 monitor Settings:
```
Comment (kasperl) [inline tools/lsp/server/server.toit:53]: Why is this a monitor?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/lsp/server/server.toit (hunk 1 above)
Comment (floitsch) [reply tools/lsp/server/server.toit:53]: Because it blocks the settings access if a `request-settings` action is in process.

Comment (kasperl) [reply tools/lsp/server/server.toit:53]: It probably would be more common to just implement this with a mutex, but it makes sense.

**Pull request merged**